pkg/dcgmexporter/system_info.go

```diff
@@ -97,21 +97,22 @@ func SetGPUInstanceProfileName(sysInfo *SystemInfo, entityId uint, profileName s
 }
 
 func SetMigProfileNames(sysInfo *SystemInfo, values []dcgm.FieldValue_v2) error {
-	notFound := false
-	err := fmt.Errorf("Cannot find match for entities:")
+	var err error
+	var errFound bool
+	errStr := fmt.Sprintf("cannot find match for entities:")
+
 	for _, v := range values {
-		found := SetGPUInstanceProfileName(sysInfo, v.EntityId, dcgm.Fv2_String(v))
-		if !found {
-			err = fmt.Errorf("%s group %d, id %d", err, v.EntityGroupId, v.EntityId)
-			notFound = true
+		if !SetGPUInstanceProfileName(sysInfo, v.EntityId, dcgm.Fv2_String(v)) {
+			errStr = fmt.Sprintf("%s group %d, id %d", errStr, v.EntityGroupId, v.EntityId)
+			errFound = true
```

Comment:
When reviewing @glowkey and @nvvfedorov please pay special attention to this change, if this improvement is ok as part of this change.

Reply:
We need to add a unit test for this part of the code.

Reply:
Adding, thanks for identifying the gap.

Reply:
@nvvfedorov Added UT for `SetMigProfileNames`.